app/controllers/api/v2/instance_hosts_controller.rb

```diff
@@ -0,0 +1,34 @@
+module Api
+  module V2
+    class InstanceHostsController < V2::BaseController
+      def resource_class
+        @resource_class ||= Host::Managed
+      end
+
+      def resource_scope(*args)
+        super.authorized(:view_hosts).joins(:infrastructure_facet).merge(::HostFacets::InfrastructureFacet.where(foreman: true))
+      end
+
+      api :PUT, '/instance/hosts/:host_id', N_("Assign a host to the Foreman instance")
+      param :host_id, :identifier_dottable
+      def update
+        # We cannot use resource scope as that is scoped only to hosts which already have the facet
+        host = ::Host::Managed.friendly.find(params[:id])
+        facet = host.infrastructure_facet || host.build_infrastructure_facet
+        facet.foreman = true
+        facet.save!
+        render status: :created, body: ''
```

Comment:
If it was already set to true, should it return HTTP OK?

Reply:
The underlying code changed a bit. I'm using `process_response`, which returns `:created` or `:ok` depending whether the request is a post or not. Since this endpoints accepts put requests, the status is always ok when it succeeds. Is this ok (pun intended)?

Reply:
It now returns `:created` when it actually changed the value, `:ok` on noop.